paperscraper/utils.py

```diff
@@ -61,26 +60,22 @@ async def _filler(self, rate_limit: float = 1):
             self.rate_limit = rate_limit
             sleep = self._get_sleep()
             updated_at = time.monotonic()
-            fraction = 0
-            extra_increment = 0
-            for i in range(self._queue.maxsize):
-                self._queue.put_nowait(i)
             while True:
-                if not self._queue.full():
-                    now = time.monotonic()
-                    increment = rate_limit * (now - updated_at)
-                    fraction += increment % 1
-                    extra_increment = fraction // 1
-                    items_2_add = int(
-                        min(
-                            self._queue.maxsize - self._queue.qsize(),
-                            int(increment) + extra_increment,
-                        )
-                    )
-                    fraction = fraction % 1
-                    for i in range(items_2_add):
-                        self._queue.put_nowait(i)
-                    updated_at = now
+                now = time.monotonic()
+                # Calculate how many tokens to add to the bucket based on elapsed time.
+                tokens_to_add = int((now - updated_at) * rate_limit)
+                # Calculate available space in the queue to avoid overfilling it.
+                available_space = self._queue.maxsize - self._queue.qsize()
+                tokens_to_add = min(
+                    tokens_to_add, available_space
+                )  # Only add as many tokens as there is space.
+
+                for _ in range(tokens_to_add):
+                    self._queue.put_nowait(
+                        None
```

Comment:
Why are we adding `None` to the queue here? Earlier we added `i`.

Can you add a comment explaining this logic?

Reply:
fixed - doesn't matter what's in queue. Added comment